cpp/core/shuffle/ShuffleWriter.h

```diff
@@ -65,6 +67,60 @@ struct ShuffleWriterOptions {
   static ShuffleWriterOptions defaults();
 };
 
+class ShuffleMemoryPool : public arrow::MemoryPool {
+ public:
+  ShuffleMemoryPool(std::shared_ptr<arrow::MemoryPool> pool) : pool_(pool) {}
+
+  arrow::MemoryPool* delegated() {
```

Comment:
Why move these functions from ShuffleWriter.cpp to here?

Reply:
these functions were removed from MemoryPoolWrapper, and the class `MemoryPoolWrapper` is unnecessary.

I agree this change because less is more.